src/main/java/soonkeatneo/duke/Parser.java

```diff
@@ -0,0 +1,81 @@
+package soonkeatneo.duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import soonkeatneo.duke.task.Deadline;
+import soonkeatneo.duke.task.Event;
+import soonkeatneo.duke.task.Todo;
+
+/**
+ * Implementation for parsing of input to be routed to specific handlers.
+ *  @author Soon Keat Neo
+ *  @version CS2103T AY20/21 Sem 1 iP
+ */
+
+public class Parser {
+    /**
+     * Handle the input and passes to the relevant methods.
+     * @param inputString User input string to be handled
+     * @param tasks Task List to be manipulated
+     * @param storage {@Storage} object to be used
+     */
+    public static void parse(String inputString, TaskList tasks, Storage storage) {
+
+        if (inputString.equals("list")) {
+            tasks.print();
+        } else if (inputString.equals("bye")) {
+            Duke.quit();
+        } else if (inputString.startsWith("done")) {
+            tasks.completeTask(inputString, storage);
+        } else if (inputString.startsWith("delete")) {
+            tasks.deleteTask(inputString, storage);
+        } else if (inputString.startsWith("todo")) {
+            try {
```

Comment:
I guess can try working on abstracting some methods out? (SLAP) 🤔 
Same for below.